R/simulations.R

```diff
@@ -381,6 +412,32 @@ sim_noise <- function(num_samples, num_vars, cov_type = "exchangeable",
   dat
 }
 
+#' @export
+#' @rdname sim_classification
+sim_logistic <- function(num_samples, eqn, correlation = 0, keep_truth = FALSE) {
+  sigma <- matrix(c(1, correlation, correlation, 1), 2, 2)
+  eqn <- rlang::get_expr(eqn)
+  dat <-
+    data.frame(MASS::mvrnorm(n = num_samples, c(0, 0), sigma)) %>%
+    stats::setNames(LETTERS[1:2]) %>%
+    dplyr::mutate(
+      .linear_pred = rlang::eval_tidy(eqn, data = .),
+      .linear_pred = as.numeric(.linear_pred),
+      .truth = stats::binomial()$linkinv(.linear_pred),
+      .rand = stats::runif(num_samples),
+      class = ifelse(.rand <= .truth, "one", "two"),
```

Comment:
this means `"one"` is the event class and that is what we want, right?

Reply:
yes, correct